airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from copy import deepcopy
+from typing import Any, Dict, Literal, Mapping
+
+from airbyte_cdk.sources.file_based.exceptions import SchemaInferenceError
+
+type_widths = {str: 0}
+
+JsonSchemaType = Literal["string"]
+supported_types = {"string"}
+
+
+def merge_schemas(schema1: Dict[str, JsonSchemaType], schema2: Dict[str, JsonSchemaType]) -> Dict[str, JsonSchemaType]:
+    """
+    Returns a new dictionary that contains schema1 and schema2.
+
+    Schemas are merged as follows
+    - If a key is in one schema but not the other, add it to the base schema with its existing type.
+    - If a key is in both schemas but with different types, use the wider type.
+    - If the type is a list in one schema but a different type of element in the other schema, raise an exception.
+    - If the type is an object in both schemas but the objects are different raise an exception.
```

Comment:
Not saying we should implement it here (i'd rather we wait till we get requests for it) but I'm curious if this is a practical or philosophical concern e.g: why not recursively merge_schemas on the objects?

Reply:
That was the original plan but during grooming we'd agreed not to support nested schemas. I took that to mean that we would just give these a generic `object` type. I made a P2 line item in the proposal for revisiting nested schemas in the future, but can add revisiting it to one of the non-CSV parser tickets if it feels higher-priority than that.